testing/fpgainfo/test_fpgainfo_c.cpp

```diff
@@ -493,6 +400,75 @@ TEST_P(fpgainfo_c_p, errors_help) {
     errors_help();
 }
 
+/**
+ * @test       parse_error_args_help
+ * @brief      Test: parse_error_args
+ * @details    When passed the help option, the function prints <br>
+ *             help message for errors subcommand.<br>
+ */
+TEST_P(fpgainfo_c_p, parse_error_args_help) {
+    char zero[20];
+    char one[20];
+    char *argv[] = { zero, one };
+
+    strcpy(zero, "fpgainfo");
+    strcpy(one, "-h");
+    EXPECT_NE(parse_error_args(2, argv), 0);
+}
+
+/**
+ * @test       errors_filter1
+ * @brief      Test: errors_filter
+ * @details    When passed with invalid arguments, the function <br>
+ *             prints help message and returns FPGA_OK. <br>
+ */
+TEST_P(fpgainfo_c_p, errors_filter1) {
+    char zero[20];
+    char one[20];
+    char two[20];
+    char *argv[] = { zero, one, two };
+
+    fpga_properties filter = NULL;
+    ASSERT_EQ(fpgaGetProperties(NULL, &filter), FPGA_OK);
+
+    strcpy(zero, "fpgainfo");
+    strcpy(one, "errors");
+    strcpy(two, "???");
+    EXPECT_EQ(errors_filter(&filter, 3, argv), FPGA_OK);
+
+    ASSERT_EQ(fpgaDestroyProperties(&filter), FPGA_OK);
+}
+
+/**
+ * @test       errors_filter2
+ * @brief      Test: errors_filter
+ * @details    When passed with argument that requires additional <br>
+ *             option, missing the option causes the fn to print <br>
+ *             help messages and return FPGA_OK. <br>
+ */
+TEST_P(fpgainfo_c_p, errors_filter2) {
+    char zero[20];
+    char one[20];
+    char two[20];
+    char *argv2[] = { zero, one };
+    char *argv3[] = { zero, one, two };
+
+    fpga_properties filter = NULL;
+    ASSERT_EQ(fpgaGetProperties(NULL, &filter), FPGA_OK);
+
+    strcpy(zero, "fpgainfo");
+    strcpy(one, "errors");
+    EXPECT_EQ(errors_filter(&filter, 2, argv2), FPGA_OK);
+
+    strcpy(two, "-c");
+    EXPECT_EQ(errors_filter(&filter, 3, argv3), FPGA_OK);
+
+    strcpy(two, "--force");
+    EXPECT_EQ(errors_filter(&filter, 3, argv3), FPGA_OK);
```

Comment:
I don't understand what these tests are doing. Are they testing the errors command or just the filter function for the errors command?

Reply:
Not entirely sure about this one. Didn't write it, only moved it. The target seems to actually be parse_error_args(), which gets called inside of errors_filter().